common/workunit/workunit.cpp

```diff
@@ -5060,6 +5114,232 @@ bool CWorkUnitFactory::restoreWorkUnit(const char *base, const char *wuid, bool
+                unzipFolder->createDirectory();
+            else
+            {//This should not happen. Just in case.
+                Owned<IDirectoryIterator> iter = unzipFolder->directoryFiles(nullptr, false, false);
+                ForEach(*iter)
+                {
+                    OwnedIFile thisFile = createIFile(iter->query().queryFilename());
+                    if (thisFile->isFile() == foundYes)
+                        thisFile->remove();
+                }
+            }
+
+            CDateTime dt;
+            dt.setNow();
+            dt.getString(unzipDateTime);
+
+            //Unzip ZAP Report
+            StringBuffer zipCommand("unzip ");
+            if (!isEmptyString(zapReportFilePath))
+                zipCommand.append(zapReportFilePath);
+            if (isEmptyString(zapReportPassword))
+                zipCommand.appendf("%s -d %s", zapReportFileName.get(), unzipDir.str());
+            else
+                zipCommand.appendf("%s -P %s -d %s", zapReportFileName.get(), zapReportPassword, unzipDir.str());
+            return (system(zipCommand.str()));
+        }
+        void buildPTreesFromZAPReport()
+        {
+            readWUXMLFileToPTree(wuTree, "ZAPReport_*.xml", false);
+            wuTree->setProp("@scope", scope);
+            wuTree->setProp("@distributedAccessToken", distributedAccessToken.str());
+
+            StringBuffer localIP, unzipDirWithIP("//");
+            IpAddress ipaddr = queryHostIP();
+            ipaddr.getIpText(localIP);
+            unzipDirWithIP.append(localIP).append(unzipDir).append(PATHSEPSTR);
+
+            //update log entries in WU XML;
+            updateWUProcessLogAttrs("Process/EclAgent/*", unzipDirWithIP, false);
+            updateWUProcessLogAttrs("Process/Thor/*", unzipDirWithIP, true);
+
+            //update QueryAssociatedFiles in WU XML;
+            updateWUQueryAssociatedFilesAttrs(localIP);
+
+            readWUXMLFileToPTree(graphProgressTree, "ZAPReport_*.graphprogress", true);
+        }
+        void setImportDebugAttribute(IWorkUnit *workunit)
+        {
+            StringBuffer attr;
+            attr.append("FromWUID=").append(wuid.get()).append(",");
+            attr.append("ImportDT=").append(unzipDateTime).append(",");
+            attr.append("ZAPReport=").append(zapReportFileName);
+            workunit->setDebugValue("imported", attr, true);
+        }
+    };
+
+    CImportWorkUnitHelper helper(zapReportFileName, user);
+    Owned<IWorkUnit> newWU = createWorkUnit(app, user, secMgr, secUser);
+    helper.getWUAttributes(newWU);
+    helper.setUNZIPDir(importDir);
+    helper.unzipZAPReport(zapReportFilePath, zapReportPassword);
+    helper.buildPTreesFromZAPReport();
+    newWU->import(helper.queryWUPTree(), helper.querygraphProgressPTree());
+    helper.setImportDebugAttribute(newWU);
+
+    Owned<IFile> zapReportFile = createIFile(zapReportFileName);
+    zapReportFile->remove();
+    return newWU.getClear();
```

Comment:
At this moment, it is not used. What I am doing is to similate the CWorkUnitFactory::createWorkUnit() which returns IWorkUnit*.

Reply:
ok, with create it makes sense - because you are about to fill in details.
Not sure it does when importing from a zap though.
I'd remove for now,
and add ability to use/test this to wutool.